ui/base-tool/build.gradle

```diff
@@ -45,5 +45,6 @@ dependencies {
     kapt "com.google.dagger:dagger-compiler:${deps.dagger}"
     kapt "com.google.dagger:hilt-compiler:${deps.hilt}"
 
+    testImplementation project(':ui:tract-renderer')
```

Comment:
If you add tests to a certain part of the app do you need to add this to the build.gradle(if it's not already there) so it will correctly run the tests?

Reply:
this is actually an unusual dependency and the only time I've done it (so far).

Normally `tract-renderer` depends on `base-tool`, and defining a normal dependency in the opposite way will introduce a dependency cycle that breaks the build.

For my unit tests in `base-tool` I wanted access to a class in `tract-renderer`, so because `test` is a separate build target/source set, I could introduce a test only dependency on the regular `tract-renderer` artifact.

so the dependency chain looks like:
`base-tool (test)` -> `tract-renderer` -> `base-tool`

Reply:
we talked about source sets and how test has it's own source set, so I'm going to resolve this